### PR TITLE
qa/cephfs: accept "force" parameter in FuseMount.umount()

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
Fixes:
https://tracker.ceph.com/issues/65876

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
- Component impact
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: qa/tasks/cephfs/fuse_mount.py [hunk 1]
```diff
@@ -308,7 +308,24 @@ def _mountpoint_exists(self):
                                       check_status=False,
                                       timeout=300).exitstatus == 0
 
-    def umount(self, lazy=True, cleanup=True):
+    def _force_before_umount(self, force=False, require_clean=False):
```
Comment (joscollin) [inline qa/tasks/cephfs/fuse_mount.py:311]: Wondering how's this getting called?

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/fuse_mount.py:311]: Nice catch, fixed.

**Review by vshankar — COMMENTED** [review]
Are _any_ tests using this additional (force/lazy) args right now?

**Comment by rishabh-d-dave** [comment]
@vshankar 
> Are _any_ tests using this additional (force/lazy) args right now?

Force, yes. Lazy, no. Code for lazy unmount was written to fix the problem described here - https://tracker.ceph.com/issues/65841. But since we are using different approach now, there's no test using this. Eventually it might be useful. Please feel free to close this PR if it's not an addition that is needed.

I had kept these commits on PR #57333 to preserve the context of these commits but were moved since Patrick wanted a separate PR for them.

**Comment by vshankar** [comment]
> @vshankar
> 
> > Are _any_ tests using this additional (force/lazy) args right now?
> 
> Force, yes. Lazy, no. Code for lazy unmount was written to fix the problem described here - https://tracker.ceph.com/issues/65841. But since we are using different approach now, there's no test using this. Eventually it might be useful. Please feel free to close this PR if it's not an addition that is needed.

Let's defer this change until we have test that require such things.

**Pull request closed without merging**